vllm/_custom_ops.py

```diff
@@ -328,22 +331,19 @@ def scaled_fp8_quant(
     else:
         output = torch.empty_like(input, dtype=torch.float8_e4m3fn)
     if scale is None:
-        scale = torch.zeros(1, device=input.device, dtype=torch.float32)
-        torch.ops._C.dynamic_scaled_fp8_quant(output, input, scale)
+        if use_per_token_if_dynamic:
+            scale = torch.empty((input.numel() // input.shape[-1], 1),
+                                device=input.device,
+                                dtype=torch.float32)
```

Comment:
Instead of `(input.numel() // input.shape[-1]` would it be better as `input.shape[0]`?

Reply:
I think the input is 3d [batch, seq_len, hidden]?

Reply:
Ok, I thought the batch and seq_len would be flattened by this point